bigquery_etl/shredder/config.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+
+"""Meta data about tables and ids for self serve deletion."""
+
+from dataclasses import dataclass
+from functools import partial
+from itertools import chain
+from typing import Optional, Tuple
+
+
+SHARED_PROD = "moz-fx-data-shared-prod"
+
+
+@dataclass(frozen=True)
+class DeleteSource:
+    """Data class for deletion request source."""
+
+    table: str
+    id_: str
+    project: str = SHARED_PROD
+
+    @property
+    def sql_table_id(self):
+        """Fully qualified table id in standard sql format."""
+        return f"{self.project}.{self.table}"
+
+
+@dataclass(frozen=True)
+class DeleteTarget:
+    """Data class for deletion request target."""
+
+    table: str
+    id_: str
+    cluster_conditions: Optional[Tuple[Tuple[str, bool]]] = None
```

Comment:
Could you document either in a comment here or in the docstring for the class what this field does? I think I'm inferring the purpose from how it's used, but it's not obvious.

Or maybe it would be good to have a separate class for the main ping clustering case. Not sure whether inheritance is possible with data classes.